src/ZfcRbac/Service/AuthorizationService.php

```diff
@@ -72,11 +73,34 @@ public function __construct(Rbac $rbac, RoleService $roleService, AssertionPlugi
      * Set an assertion
      *
      * @param string|PermissionInterface         $permission
-     * @param string|callable|AssertionInterface $assertion
+     * @param string|callable|array|AssertionInterface $assertion
      * @return void
      */
     public function setAssertion($permission, $assertion)
     {
+        // if is name of the assertion, retrieve an actual instance from assertion plugin manager
+        if (is_string($assertion)) {
+            $assertion = $this->assertionPluginManager->get($assertion);
+        } elseif (is_array($assertion)) { // else if multiple assertion definition, create assertion set.
```

Comment:
What about the following approach?
- `assertion_map` config key accepts only `string => string` (permission name => assertion name)
- Another config key for `assertion_sets` that accepts `string => array` (assertion name => assertion config)
- A default `AssertionSetFactory` implementation that reads the `assertion_sets` config key and builds `AssertionSet`
- `AuthorizationService` always fetch assertions from `AssertionPluginManager`